samcli/lib/hook/hook_wrapper.py

```diff
@@ -0,0 +1,176 @@
+    metadata_loc = hook.prepare("path/to/iac_project", "path/to/output", True, "path/to/logs")
+    ```
+    """
+
+    _config: Optional[HookPackageConfig]
+
+    _INTERNAL_PACKAGES_ROOT = Path(__file__).parent / ".." / ".." / "hook_packages"
+
+    def __init__(self, hook_package_id: str):
+        """
+        Parameters
+        ----------
+        hook_package_id: str
+            Hook package ID
+        """
+        self._config = None
+        self._load_hook_package(hook_package_id)
+
+    def prepare(
+        self,
+        output_dir_path: str,
+        iac_project_path: Optional[str] = None,
+        debug: bool = False,
+        logs_path: Optional[str] = None,
+        aws_profile: Optional[str] = None,
+        aws_region: Optional[str] = None,
+    ) -> str:
+        """
+        Run the prepare hook to generate the IaC Metadata file.
+
+        Parameters
+        ----------
+        output_dir_path: str
+            the path where the hook can create the generated Metadata files. Required
+        iac_project_path: str
+            the path where the hook can find the TF application. Default value in current work directory.
+        debug: bool
+            True/False flag to tell the hooks if should print debugging logs or not. Default is False.
+        logs_path: str
+            File path where the hooks will write its logs. Default is None (no need to write logs)
+        aws_profile: str
+            AWS profile to use. Default is None (use default profile)
+        aws_region: str
+            AWS region to use. Default is None (use default region)
+
+        Returns
+        -------
+        str
+            Path to the generated IaC Metadata file
+        """
+        params = {
+            "IACProjectPath": iac_project_path if iac_project_path else str(Path.cwd()),
+            "OutputDirPath": output_dir_path,
+            "Debug": debug,
+        }
+        if logs_path:
+            params["LogsPath"] = logs_path
+        if aws_profile:
+            params["Profile"] = aws_profile
+        if aws_region:
+            params["Region"] = aws_region
+
+        output = self._execute("prepare", params)
+        metadata_file_loc = output.get("IACApplications", {}).get("MainApplication", {}).get("Metadata")
+        if not metadata_file_loc:
+            raise InvalidHookWrapperException("Metadata file path not found in the prepare hook output")
+        return cast(str, metadata_file_loc)
+
+    def _load_hook_package(self, hook_package_id: str) -> None:
+        """Find and load hook package config with given hook package ID
+
+        Parameters
+        ----------
+        hook_package_id: str
+            Hook package ID
+        """
+        # locate hook package from internal first
+        for child in self._INTERNAL_PACKAGES_ROOT.iterdir():
+            try:
+                hook_package_config = HookPackageConfig(child)
+                if hook_package_config.package_id == hook_package_id:
```

Comment:
let's do not use the Config.json package Id as the `hook-pacakge-id` parameter value, because in future if we will move to 3rd party hooks, and we implement the `sam hooks install` command, the customer may be able to customize the hook package id to solve the issue if there are multiple hooks that were developed and their owners use the same package id.
So, for now for the internal use, my assumption was to use the directory name as the hook_package_id.